examples/api/lib/widgets/overlay/overlay_portal.0.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for OverlayPortal
+
+import 'package:flutter/material.dart';
+
+void main() => runApp(const MyApp());
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Code Sample',
+      home: Scaffold(
+        appBar: AppBar(title: const Text('OverlayPortal Example')),
+        body: const Center(child: ClickableTooltipWidget()),
+      ),
+    );
+  }
+}
+
+class ClickableTooltipWidget extends StatefulWidget {
+  const ClickableTooltipWidget({super.key});
+
+  @override
+  State<StatefulWidget> createState() => ClickableTooltipWidgetState();
+}
+
+class ClickableTooltipWidgetState extends State<ClickableTooltipWidget> {
+  final OverlayPortalController _tooltipController = OverlayPortalController();
+
+  @override
+  Widget build(BuildContext context) {
+    return TextButton(
+      onPressed: _tooltipController.toggle,
+      child: DefaultTextStyle(
+        style: DefaultTextStyle.of(context).style.copyWith(fontSize: 50),
+        child: OverlayPortal(
+          controller: _tooltipController,
+          overlayChildBuilder: (BuildContext context) {
+            return const Positioned(
+              right: 50,
+              bottom: 50,
+              child: ColoredBox(
+                color: Colors.amberAccent,
+                child: Text('tooltip'),
+              ),
+            );
+          },
+          child: const Text('Press to show/hide tooltip'),
+        ),
+      ),
+    );
```

Comment:
Just naively looking at this API, it seems to me like it could be much more declarative if it dropped the OverlayPortalController.  Could you control show/hide simply like this?

```dart
    return TextButton(
      onPressed: () {
        setState(() {
          _tooltipVisible = !_tooltipVisible;
        });
      },
      child: DefaultTextStyle(
        style: DefaultTextStyle.of(context).style.copyWith(fontSize: 50),
        child: OverlayPortal(
          overlayChildBuilder: (BuildContext context) {
            if (!_tooltipVisible) {
              return null;
            }
            return const Positioned(
              right: 50,
              bottom: 50,
              child: ColoredBox(
                color: Colors.amberAccent,
                child: Text('tooltip'),
              ),
            );
          },
          child: const Text('Press to show/hide tooltip'),
        ),
      ),
    );
```

Or maybe even scrap the OverlayPortal altogether if that's better?

```dart
    const Text text = Text('Press to show/hide tooltip');
    final OverlayPortal portal = OverlayPortal(
      overlayChildBuilder: (BuildContext context) {
        return const Positioned(
          right: 50,
          bottom: 50,
          child: ColoredBox(
            color: Colors.amberAccent,
            child: Text('tooltip'),
          ),
        );
      },
      child: text,
    );

    return TextButton(
      onPressed: () {
        setState(() {
          _tooltipVisible = !_tooltipVisible;
        });
      },
      child: DefaultTextStyle(
        style: DefaultTextStyle.of(context).style.copyWith(fontSize: 50),
        child: _toolbarIsVisible ? overlayPortal : text,
      ),
    );
```

Reply:
The purpose of introducing a controller is to get the timestamp of the user event that caused the overlay child to appear, and use the timestamp as the z-index to put different overlay children into the correct paint order. See https://github.com/flutter/flutter/pull/105335#issuecomment-1207443527